src/v2/Apps/Auctions/Components/WorksByArtistsYouFollowRail/WorksByArtistsYouFollowRail.tsx

```diff
@@ -28,6 +21,10 @@ const WorksByArtistsYouFollowRail: React.FC<WorksByArtistsYouFollowRailProps> =
   const { contextPageOwnerType } = useAnalyticsContext()
   const contextModule = tabTypeToContextModuleMap.worksByArtistsYouFollow
 
+  if (viewer.saleArtworksConnection.edges.length === 0) {
```

Comment:
All good points! but....

Bailing higher up: Because of Relay data masking, data for a component needs to be resolved within that component. I mean, one could def define a fully-resolved schema up above, but then we'd be defining it twice while _also_ breaking component-data encapsulation patterns.

Moving to top of file: One would think, yah? Unfortunately that breaks the [rule of hooks](https://www.benmvp.com/blog/conditional-react-hooks). Its some React magic that i'm not too fond of, given that pre-hooks React didn't have much of that.

Reply:
Darn, oh well!! 😝